code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -113,43 +113,97 @@
 
 /obj/item/gun/energy/lwap
 	name = "LWAP laser sniper"
-	desc = "A highly advanced laser sniper that does more damage the farther away the target is, but fires slowly."
+	desc = "A highly advanced laser sniper that does more damage the farther away the target is, but fires slowly. Comes with a super advanced scope, which can highlight threats through walls, and pierce one object, after being deployed for a while."
 	icon_state = "esniper"
 	item_state = null
 	w_class = WEIGHT_CLASS_BULKY
 	force = 12
-	flags =  CONDUCT
+	flags = CONDUCT
 	slot_flags = SLOT_BACK
 	can_holster = FALSE
 	weapon_weight = WEAPON_HEAVY
 	origin_tech = "combat=6;magnets=6;powerstorage=4"
-	ammo_type = list(/obj/item/ammo_casing/energy/laser/sniper)
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/sniper, /obj/item/ammo_casing/energy/laser/sniper/pierce)
 	zoomable = TRUE
 	zoom_amt = 7
 	shaded_charge = TRUE
+	/// Is the scope fully online or not?
+	var/scope_active = FALSE
+	var/stored_dir
+
+/obj/item/gun/energy/lwap/Initialize(mapload, ...)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/gun/energy/lwap/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+
+/obj/item/gun/energy/lwap/zoom(mob/living/user, forced_zoom)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	stored_dir = H.dir
+	if(scope_active && !zoomed)
+		select_fire(H)
+		H.remove_status_effect(STATUS_EFFECT_LWAPSCOPE)
+		scope_active = FALSE
+	else if(do_after(user, 3 SECONDS, target = src) && zoomed)
+		if(zoomed && !scope_active) //We check after to be sure.
+			scope_active = TRUE
+			to_chat(user, "<b><span class='robot'>SCOPE_CREEPER_[rand(1, 9999)] Online.</span></b>")
+			select_fire(H)
+			H.apply_status_effect(STATUS_EFFECT_LWAPSCOPE, stored_dir)
+
+/obj/item/gun/energy/lwap/process()
+	. = ..()
+	if(!isliving(loc))
+		return
+	var/mob/living/M = loc
+	if(world.time - M.last_movement <= 21 && zoomed) //Not perfect, will cause moving just before scoping to not be happy, open for a better solution.
```

Comment:
Magic number here `21`

Reply:
do we have a define for a process cycle, or should I just do 2.1 seconds.

Reply:
Changed to 2.1 seconds + a comment

Reply:
2.1 SECONDS sounds a bit arbitrary, giving it a define/variable which you can name might help explain why you picked 2.1 SECONDS